app/jobs/concerns/maintenance_tasks/task_job_concern.rb

```diff
@@ -57,11 +57,18 @@ def build_enumerator(_run, cursor:)
       when CSV
         JobIteration::CsvEnumerator.new(collection).rows(cursor: cursor)
       else
-        raise ArgumentError, <<~MSG.squish
-          #{@task.class.name}#collection must be either an
-          Active Record Relation, ActiveRecord::Batches::BatchEnumerator,
-          Array, or CSV.
-        MSG
+        if collection.respond_to?(:to_enum)
```

Comment:
This conflicts with `to_enum` in the standard library. Do we want to make a different method to avoid needing to rescue from type errors?